app/Models/Beatmapset.php

```diff
@@ -454,6 +454,40 @@ public static function search(array $params = [])
             : [];
     }
 
+    public static function latestRankedOrApproved($count = 5)
+    {
+        // TODO: add filtering by game mode after mode-toggle UI/UX happens
+
+        return Cache::remember("beatmapsets_latest_{$count}", 60, function () use ($count) {
+            // We union here so mysql can use indexes to speed this up
+            $ranked = self::ranked()->active()->orderBy('approved_date', 'desc')->limit($count);
```

Comment:
is there reason to not just use `rankedOrApproved`? Seems like using same indexes.